torch/csrc/lazy/core/ir.cpp

```diff
@@ -46,21 +51,37 @@ bool Node::enableDynamicShape() {
   return enabled || FLAGS_ltc_enable_dynamic_shapes;
 }
 
+size_t Node::NextNodeListIndex() {
+  // Tracing is done in a single thread, so no need to use atomic here.
```

Comment:
i think this is good enough for prototyping, but at some point we'd need a design that is thread safe in case someone did use multiple threads..